internal/types/binary.go

```diff
@@ -35,3 +41,42 @@ type Binary struct {
 	Subtype BinarySubtype
 	B       []byte
 }
+
+func BinaryFromArray(values *Array) (*Binary, error) {
```

Comment:
Our linter for that is broken right now (should be fixed by 1.18 release), but: every exported function should have a documentation comment.

It is somewhat hard to understand what conversion from array to binary even mean